### ISSUE TITLE
Audio programming

### ISSUE THREAD

**Issue by SirPython** [Cardshifter/Cardshifter#368]
It would be nice to integrate some audio programming into the client so, depending on the state of the game, the music could intensify and relax accordingly.

> [But imagine you had music that was programmatically generated? Like the intensity, tempo, key or whatever can change based on game events!](http://chat.stackexchange.com/transcript/message/23454423#23454423)

**Comment by marcandregirard** [comment]
Just make sure it's off by default and volume can be set and I am personally all happy ;)
http://chat.stackexchange.com/transcript/message/23584928#23584928

I'm quoting skiwi here since I find this is very important. I mostly play with friends online with a skype call or another program. Most of the time, I will mute the music of a game so that I can hear my friend. Keep this in mind when/if you add this feature to the game.

**Comment by Phrancis** [comment]
Aside from the possibility of using [ChucK audio programming language](https://en.wikipedia.org/wiki/ChucK) _(which at this time is undecided)_ here are some more client-side options and links.

**libgdx API** : [Audio](https://libgdx.badlogicgames.com/nightlies/docs/api/com/badlogic/gdx/Audio.html) || [Music](https://libgdx.badlogicgames.com/nightlies/docs/api/com/badlogic/gdx/audio/Music.html) || [More...](https://www.google.com/search?q=libgdx+audio+music)

**jQuery** : [Native](https://www.google.com/search?q=jquery+audio&ie=utf-8&oe=utf-8) || [jPlayer](http://jplayer.org/) || [More](https://www.google.com/search?q=jquery+audio+music)

**AngularJS** : [ngAudio](http://danielstern.github.io/ngAudio/#/)

**HTML 5** : [`<audio>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/audio) element

**Comment by SirPython** [comment]
@Phrancis The ngAudio is cool, but it doesn't really allow for a dynamic change in the audio, unless just simple changes in dynamics and tempo are enough.

The same applies to the audio element.

**Comment by SirPython** [comment]
[This site](https://www.javascripting.com/audio/) has a list of audio library for JavaScript.

**Comment by Phrancis** [comment]
@SirPython Yeah I think I agree. Likely jQuery will be a better tool for this job or vanilla JS.

**Comment by SirPython** [comment]
@Phrancis I read on about that audio tag, and it looks like there is something called the [Web Audio API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Audio_API) would could possibly allow for dynamic sound/music modification.
